code/game/machinery/computer/security.dm

```diff
@@ -34,12 +22,89 @@
 			O.forceMove(src)
 			scanner = O
 			to_chat(user, "You insert [O].")
+			update_static_data(usr)
 
 	. = ..()
 
+/obj/structure/machinery/computer/secure_data/ui_static_data(mob/user)
+	. = ..()
+
+	if(!allowed(usr))
+		to_chat(user, SPAN_WARNING("Access denied."))
+		return
+
+	// Pass player data, only for frontend state
+	var/mob/living/carbon/human/U = usr
+	var/user_data = list("name" = U.get_authentification_name(), "rank" = U.get_assignment())
+	.["user_data"] = user_data
```

Comment:
access and user data dont belong here, use ui_act to double check a user can do an action and pass anything about a current user with ui_data

Reply:
UI open and ui_act already checked for permission, it was redundant here. I removed user_data as well, cause it wasn't needed in UI anyway.